src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3040,42 +3048,70 @@ void CodeGen::genCodeForCpBlkUnroll(GenTreeBlk* node)
             }
         }
 
-        // TODO-CQ-XArch: On x86 we could copy 8 byte at once by using MOVQ instead of four 4 byte MOV stores.
-        // On x64 it may also be worth copying a 4/8 byte remainder using MOVD/MOVQ, that avoids the need to
-        // allocate a GPR just for the remainder.
-    }
-
-    if (size > 0)
-    {
-        regNumber tempReg = node->GetSingleTempReg(RBM_ALLINT);
-
-        for (unsigned regSize = REGSIZE_BYTES; size > 0; size -= regSize, srcOffset += regSize, dstOffset += regSize)
+        if (size > 0)
         {
-            while (regSize > size)
-            {
-                regSize /= 2;
-            }
+            // Copy the remainder by moving the last regSize bytes of the buffer
+            unsigned remainder = regSize - size;
```

Comment:
Is it worth an assert that `(remainder <= regSize)`?

Reply:
Sure, that makes sense. I'll add that

Reply:
> I don't expect this to have a perf difference, although I'd have to benchmark it to be sure. It seemed easier/cleaner to just reuse the register and regSize that was setup pre-copy. Let me know if you think it's better to break up cases like this instead.

I think that the change to support is it kind of small, you can use the same register just 

```suggestion
            if ((regSize > XMM_REGSIZE_BYTES) && (size <= XMM_REGSIZE_BYTES))
            {
              regSize  = XMM_REGSIZE_BYTES;
            }
            unsigned remainder = regSize - size;
```

Reply:
`size` really means `remainingSize` if I understand correctly, so I would at least rename remainder because the real is `size`.
```suggestion
            unsigned shiftBack = regSize - size; // so we could use another full move [totalSize - regSize; totalSize).
```